pkg/database/gorm.go

```diff
@@ -1,18 +1,54 @@
 package database
 
 import (
+	"fmt"
 	"strconv"
 	"time"
 
 	"github.com/DATA-DOG/go-txdb"
+	gormtrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/gorm.io/gorm.v1"
 	"gorm.io/driver/mysql"
 	"gorm.io/gorm"
 )
 
 const testEnv = "test"
 
-// NewConnection returns a new Gorm database connection.
+// NewInstrumentedConnection returns a new instrumented Gorm database connection.
+func NewInstrumentedConnection(config *Config, environment, appName string) (*gorm.DB, error) {
+	return newConnection(config, environment, appName, true)
+}
+
+// NewConnection returns a new uninstrumented Gorm database connection.
 func NewConnection(config *Config, environment string) (*gorm.DB, error) {
+	return newConnection(config, environment, "", false)
+}
+
+func newConnection(
+	config *Config, environment, appName string, instrumented bool,
+) (*gorm.DB, error) {
+	dialector := getDialectorFromConfig(config, environment)
+
+	var db *gorm.DB
+	var err error
+	if instrumented {
+		serviceName := fmt.Sprintf("%s-mysql", appName)
+		if db, err = gormtrace.Open(dialector, nil, gormtrace.WithServiceName(serviceName)); err != nil {
+			return nil, err
+		}
+	} else {
+		if db, err = gorm.Open(dialector); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
One instrumented connection will also simplify the flow/logic here as well.

It will be much more straightforward. ⏩